lib/NodeHelpers.js

```diff
@@ -130,3 +131,44 @@ function getLocalMatrix4(node, result) {
 
     return Matrix4.fromTranslationQuaternionRotationScale(translation, rotation, scale, result);
 }
+
+var packedParametersScratch = {
+    meshes : undefined,
+    parameters : undefined,
+    primitiveFunction : undefined
+};
+
+// Perform an operation on each primitive in a scene.
+// PrimitiveFunction should expect primitive, a meshPrimitiveID, some parameters, and the node itself
+function forEachPrimitiveInScene(gltf, scene, primitiveFunction, parameters) {
+    var rootNodeNames = scene.nodes;
+    var allNodes = gltf.nodes;
+    packedParametersScratch.meshes = gltf.meshes;
+    packedParametersScratch.parameters = parameters;
+    packedParametersScratch.primitiveFunction = primitiveFunction;
+    for (var nodeID in rootNodeNames) {
+        if (rootNodeNames.hasOwnProperty(nodeID)) {
+            var rootNodeName = rootNodeNames[nodeID];
+            depthFirstTraversal(allNodes[rootNodeName], allNodes, forEachPrimitiveInNode, packedParametersScratch);
+        }
+    }
+}
+
+function forEachPrimitiveInNode(packedParameters, node) {
+    var meshes = packedParameters.meshes;
+    var parameters = packedParameters.parameters;
+    var primitiveFunction = packedParameters.primitiveFunction;
+
+    var meshIDs = node.meshes;
+    if (!defined(meshIDs)) {
+        return;
+    }
+    for (var i = 0; i < meshIDs.length; i++) {
```

Comment:
Here, below, and throughout - unless length could change, extract it from the loop, e.g.,
```javascript
var length = meshIDs.length;
// ...
```